src/btree/bt_sync.c

```diff
@@ -109,8 +109,19 @@ __sync_ref_is_obsolete(WT_SESSION_IMPL *session, WT_REF *ref)
 {
     WT_ADDR *addr;
 
-    addr = ref->addr;
+    /* Ignore root pages as they can never be deleted. */
+    if (__wt_ref_is_root(ref))
+        return false;
+
+    /* Guard against marking an in-memory page to be deleted. */
+    if (ref->state != WT_REF_DISK || !WT_REF_CAS_STATE(session, ref, WT_REF_DISK, WT_REF_LOCKED))
+        return false;
+
+    /* Ignore internal pages, these are taken care during reconciliation. */
```

Comment:
Comment is missing a word, "these are taken care of".

Reply:
Corrected.